fs/memoryfs.py

```diff
@@ -342,7 +342,8 @@ def _get_dir_entry(self, dir_path):
 
     def close(self):
         # type: () -> None
-        self.root = None
+        if not self._closed:
+            del self.root
```

Comment:
Mypy complains when assigning None, rather than modify the type of `root` just del it.